christfellowshipchurch/web-app
# [Events] Sandbox for new Events Layout and Chat POC

# About
Adds a sandbox version of the existing Event page, and integrates a crude chat POC along with general layout/structural changes to support it.

**TL;DR on the Sandbox configuration**
Files named `.sandbox.js` in the **web-app** are loaded _only in the beta_ environment.
This is separate altogether from the feature flagging system, and is a simple way to maintain old and new versions of a page without worrying about end-user visibility in the mean time.

After this PR is merged, the current production CF events page should remain unchanged — but the same page on beta.christfellowship.church should start using the new `.sandbox.js` versions of components.

# Tradeoffs/Caveats
**Responsive handling** and the mobile view in general were ignored. No real effort was spent massaging padding/spacing details or what happens on smaller viewports.

**Stream.io chat component styling** is still a bit of an unknown. I added a [stream-chat-react-overrides.css](https://github.com/christfellowshipchurch/web-app/pull/149/files#diff-c0eed21003356fc2d1e69876c214907d) file for now, to allow hard coding the height for scaffolding purposes. This solution is far from ideal, but just needs a closer look.

**User authentication** and general Stream client management is _very_ crude here. Next PR will have proper initialization and management of Stream client, with loading/connection state handlers/token fetching/etc. (Bulk of this is on Caleb's branch already, and/or parallels mobile app).

# Test Instructions
You'll need to update your local `.env` file to include a new `REACT_APP_STREAM_API_KEY` value.
The stream.io credentials/keys should match what your local API uses.

1. Start web app and api locally
1. Go to an event page and verify you can see the chat off to the side and general WIP presentation (screenshots below)
1. All messages you send will be as **Yoda**, whether you're logged in or out
1. In the PR CI/CD area, find the **netlify/deploy-preview** jobs and smoke test that the new development page only shows on beta.christfellowship.church


# Screenshots

|Large|Medium|Small|
|---|---|---|
|![CleanShot 2020-08-28 at 15 22 48@2x](https://user-images.githubusercontent.com/1812955/91767413-19eab000-ebaa-11ea-9ae1-76e83a335b4f.jpg)|![CleanShot 2020-08-28 at 15 23 09@2x](https://user-images.githubusercontent.com/1812955/91767424-1d7e3700-ebaa-11ea-8197-e02390aeb594.jpg)|![CleanShot 2020-08-28 at 15 23 18@2x](https://user-images.githubusercontent.com/1812955/91767426-1e16cd80-ebaa-11ea-8293-78ea3fbd4ede.jpg)|


# Tests
